lib/workers/repository/process/limits.ts

```diff
@@ -40,22 +40,32 @@ export async function getPrHourlyRemaining(
   return 99;
 }
 
-export function getConcurrentPrsRemaining(
+export async function getConcurrentPrsRemaining(
   config: RenovateConfig,
   branches: BranchConfig[]
-): number {
+): Promise<number> {
   if (config.prConcurrentLimit) {
-    logger.debug(`Enforcing prConcurrentLimit (${config.prConcurrentLimit})`);
-    let currentlyOpen = 0;
-    for (const branch of branches) {
-      if (branchExists(branch.branchName)) {
-        currentlyOpen += 1;
-      }
+    logger.debug(`Calculating prConcurrentLimit (${config.prConcurrentLimit})`);
+    try {
+      const branchList = branches.map(({ branchName }) => branchName);
+      const prList = await Promise.all(
+        branchList.map((branchName) => platform.getBranchPr(branchName))
+      );
+      const openPrs = prList
+        .filter(Boolean)
```

Comment:
🤔  Do we really need this?

Reply:
Not sure about "really" part of your question, but `getBranchPr` can possibly return `null` according to its signature. To make it more bulletproof, I would even replace `Promise.all` with manual loop with try-catch checks.

Reply:
ok, it shouldn't really matter if we do this sequencially, as the pr list would be fetch only once from server.

Reply:
I think this logic makes sense to filter out any `null` responses for when PRs don't exist for the branch yet